### PR TITLE
Feature/selc 147

### PR DESCRIPTION
**Pull request by rodoviti**

### CODE REVIEW

File: core/src/main/java/it/pagopa/selfcare/product/core/security/UserDetailsServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+//@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    @Override
+    public UserDetails loadUserByUsername(String s) throws UsernameNotFoundException {
+        return null;
```
Comment (antonioT90) [inline core/src/main/java/it/pagopa/selfcare/product/core/security/UserDetailsServiceImpl.java:12]: put a TODO to rember to fix it

File: core/src/test/java/it/pagopa/selfcare/product/core/ProductServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                .thenAnswer(invocationOnMock -> new Product("logo", "title", "description", "urlPublic", "urlBO"));
+        // when
+        Product product = productService.createProduct(prod);
+        // then
+        assertFalse(product.toString().isEmpty());
+    }
+
+    @Test
+    void deleteProducts() {
+
```
Comment (antonioT90) [inline core/src/test/java/it/pagopa/selfcare/product/core/ProductServiceImplTest.java:68]: why this is empty? if not to do, remove it, if to do later put a TODO on it

File: integration/dao/src/main/resources/config/dao-config.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+spring.data.mongodb.database=selc_dashboard
```
Comment (antonioT90) [inline integration/dao/src/main/resources/config/dao-config.properties:1]: These properties should be configurable through ENV

File: web/src/main/java/it/pagopa/selfcare/product/web/model/ProductResource.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    private String title;
+    @Schema(description = "${swagger.product.description}")
+    private String description;
+    @Schema(description = "${swagger.product.urlPublic}")
+    private String urlPublic;
+    @Schema(description = "${swagger.product.urlBO}")
+    private String urlBO;
+
+
+    public static ProductResource create(Product product) {
```
Comment (antonioT90) [inline web/src/main/java/it/pagopa/selfcare/product/web/model/ProductResource.java:24]: write a test to verify that:
* given a Product with all properties setted, every field of the ProductResource has the expected value
* the field names of Product class are the same of ProductResource (eventually with some controlled exceptions)

File: web/src/main/java/it/pagopa/selfcare/product/web/model/UpdateProductDto.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    private String title;
+    @Schema(description = "${swagger.product.model.description}")
+    private String description;
+    @Schema(description = "${swagger.product.model.urlPublic}")
+    private String urlPublic;
+    @Schema(description = "${swagger.product.model.urlBO}")
+    private String urlBO;
+
+
+    public static Product toEntity(UpdateProductDto updateProductDto) {
```
Comment (antonioT90) [inline web/src/main/java/it/pagopa/selfcare/product/web/model/UpdateProductDto.java:22]: write a test to verify that:

given a Product with all properties setted, every field of the ProductResource has the expected value
the field names of Product class are the same of ProductResource (eventually with some controlled exceptions)

File: web/src/main/java/it/pagopa/selfcare/product/web/security/CustomAuthProvider.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
+package it.pagopa.selfcare.product.web.security;
+
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+
+
+//@Component
+public class CustomAuthProvider implements AuthenticationProvider {
```
Comment (antonioT90) [inline web/src/main/java/it/pagopa/selfcare/product/web/security/CustomAuthProvider.java:9]: put a TODO here to remember to fix

File: web/src/main/resources/config/jwt.properties [hunk 7]
```diff
@@ -0,0 +1,2 @@
+jwt.secret=bXlTZWNyZXRLZXk=
```
Comment (antonioT90) [inline web/src/main/resources/config/jwt.properties:1]: These values should be configurable by ENV vars

File: integration/dao/src/main/resources/config/dao-config.properties (hunk 3 above)
Comment (rodoviti) [reply integration/dao/src/main/resources/config/dao-config.properties]: done

File: web/src/main/java/it/pagopa/selfcare/product/web/security/CustomAuthProvider.java (hunk 6 above)
Comment (rodoviti) [reply web/src/main/java/it/pagopa/selfcare/product/web/security/CustomAuthProvider.java]: added TODO

File: web/src/main/resources/config/jwt.properties (hunk 7 above)
Comment (rodoviti) [reply web/src/main/resources/config/jwt.properties]: added as ENV VAR

File: web/src/main/java/it/pagopa/selfcare/product/web/model/ProductResource.java (hunk 4 above)
Comment (rodoviti) [reply web/src/main/java/it/pagopa/selfcare/product/web/model/ProductResource.java:24]: this is not a "strictly necessary" test, so mi suggestion is to create a new jira task to tracks it

File: web/src/main/java/it/pagopa/selfcare/product/web/model/UpdateProductDto.java (hunk 5 above)
Comment (rodoviti) [reply web/src/main/java/it/pagopa/selfcare/product/web/model/UpdateProductDto.java:22]: this is not a "strictly necessary" test, so mi suggestion is to create a new jira task to tracks it

Comment (rodoviti) [reply web/src/main/java/it/pagopa/selfcare/product/web/model/UpdateProductDto.java:22]: here the [jira task](https://pagopa.atlassian.net/browse/SELC-162)

**Pull request merged**